src/dense/linearsystemsolver.jl

```diff
@@ -121,7 +121,7 @@ for type in magmaTypeList
             # end
             lda  = max(1,stride(A,2))
             ldb  = max(1,stride(B,2))
-            ipiv = similar(Matrix(A), Int, n)
+            ipiv = similar(Matrix(A), Int32, n)
```

Comment:
by default Julia will use `Int` for sizes which is defined according to to system, why change this to hard-coded Int32?

Reply:
Because it is coded as  `int` C type in magma with is 32 bits and the array `ipiv` is not passed correctly.